//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package kx.grpc_server;

// A request message containing a word to be translated.
message GenerateKeyPairRequest {

}

message GenerateKeyPairReply {
  required string encrypted_private_key = 1;
  required string public_key = 2;
}


message SignMessageRequest {
  required string message = 1;
  required string encrypted_private_key = 2;
}

message SignMessageReply {
  required string signature = 1;
}


service SecureEnclave {
  rpc GenerateKeyPair(GenerateKeyPairRequest) returns (GenerateKeyPairReply) {

  }

  rpc SignMessage(SignMessageRequest) returns (SignMessageReply) {

  }
}